src/commands/createNewProject/NewProjectLanguageStep.ts

```diff
@@ -44,18 +49,22 @@ export class NewProjectLanguageStep extends AzureWizardPromptStep<IProjectWizard
             { label: ProjectLanguage.CSharp, data: { language: ProjectLanguage.CSharp } },
             { label: ProjectLanguage.Python, data: { language: ProjectLanguage.Python } },
             { label: ProjectLanguage.Java, data: { language: ProjectLanguage.Java } },
-            { label: ProjectLanguage.Ballerina, data: { language: ProjectLanguage.Ballerina } },
             { label: ProjectLanguage.PowerShell, data: { language: ProjectLanguage.PowerShell } },
-            { label: localize('customHandler', 'Custom Handler'), data: { language: ProjectLanguage.Custom } }
+            { label: localize('customHandler', 'Custom Handler'), data: { language: ProjectLanguage.Custom } },
+            { label: localize('selfHostedMCPServer', 'Self-hosted MCP server'), data: { language: ProjectLanguage.SelfHostedMCPServer } }
         ];
 
+        if (getWorkspaceSetting(showBallerinaProjectCreationSetting)) {
+            languagePicks.push({ label: ProjectLanguage.Ballerina, data: { language: ProjectLanguage.Ballerina } });
+        }
+
         if (context.languageFilter) {
             languagePicks = languagePicks.filter(p => {
                 return context.languageFilter?.test(p.data.language);
             });
         }
 
-        const options: QuickPickOptions = { placeHolder: localize('selectLanguage', 'Select a language') };
+        const options: QuickPickOptions = { placeHolder: localize('selectProjectType', 'Select a project type') };
```

Comment:
Based on this, perhaps we could eventually changes variable and enum names to reflect project type or similar